src/ml_gym/util/util.py

```diff
@@ -61,6 +106,154 @@ def create_system_info() -> Dict:
+                exp_config (dict): Experiment configuration.
+                gs_config (dict): Grid Search configuration.
+        :returns:
+                model_card (ModelCard): Model card object to be converted into json file.
+        """
+        
+        def update_model_details(grid_search_id: str, gs_config: dict, model) -> ModelDetails:
+            """
+            Function to initialize ModelDetails object.
+            :params:
+                    grid_search_id (str): Grid Search ID created for the run.
+                    gs_config (dict): Grid Search configuration.
+            :returns:
+                    obj (ModelDetails): initialized model details object.
+            """
+            try:
+                if model != None:
+                    pytorch_total_params = sum(p.numel() for p in model.parameters() if p.requires_grad)
+                else:
+                    pytorch_total_params = 0
+                train_date = datetime.now().strftime("%d-%m-%Y")
+                model_info = gs_config["model_info"]
+                return ModelDetails(model_description = model_info["model_description"], model_version = model_info["model_version"], grid_search_id = grid_search_id, train_date = train_date, source_repo = model_info["source_repo"], train_params= pytorch_total_params)
+            except Exception as e:
+                print("Error while fetching Model Details for Model card.", e)
+                return ModelDetails()
+        
+        def update_dataset_details(exp_config: dict) -> DatasetDetails:
+            """
+            Function to initialize DatasetDetails object.
+            :params:
+                    exp_config (dict): Experiment configuration.
+            :returns:
+                    obj (DatasetDetails): initialized dataset details object.
+            """
+            try:
+                dataset_splits = {"split_config": exp_config["dataset_iterators"]["config"]["split_configs"], 
+                              "splits_percentage": exp_config["splitted_dataset_iterators"]["config"]["split_configs"]}
+            
+                return DatasetDetails(considered_dataset = exp_config["dataset_iterators"]["config"]["dataset_identifier"], dataset_splits = dataset_splits)
+            except Exception as e:
+                print("Error while fetching Dataset Details for Model card.", e)
+                return DatasetDetails()
+        
+        def update_training_details(exp_config: dict, gs_config: dict) -> TrainingDetails:
+            """
+            Function to initialize TrainingDetails object.
+            :params:
+                    exp_config (dict): Experiment configuration.
+            :returns:
+                    obj (TrainingDetails): initialized training details object.
+            """
+            try:
+                hyperparams = {}
+                hyperparams["optimizer"] = exp_config["optimizer"]["config"]["params"]
```

Comment:
why did you hardcode this for optimizer only? There are various places where we might have a parameter sweep. E.g., different loss functions, different model paramters (e.g., activation functions), different seeds etc. 

So as I suggested in our previous meeting, we need to create this dynamically without hardcoding to certain components. We can get the information dynamically from the config of a single experiment (concrete value for a sweep) and the grid search yaml (contains all the sweeps and the ranges).

Reply:
Well, this is not actually hard-coded but this was done to get the params of a particular sweep from the experiment config. Since we wanted to change the JSON structure I have done this to get something like this: 
"hyperparams": {
            "optimizer": {
                "lr": 1
            }

Reply:
this information is being fetched dynamically but put into a static JSON structure for the model card. I did not compare it with the sweeps in grid search yaml, which I will do now.

Reply:
`exp_config["optimizer"]["config"]["params"]` the path is still harcoded. There might be configs that have a path like `exp_config["optimizer"]["config_2"]["paramsXY"]`.  We need to make sure that we get the path information from the grid search config

Reply:
ok will try to  see how to fix this